os/fs/smartfs/smartfs_smart.c

```diff
@@ -1683,26 +1731,53 @@ static int smartfs_mkdir(struct inode *mountpt, const char *relpath, mode_t mode
 		ret = -EEXIST;
 		goto errout_with_semaphore;
 	} else if (ret == -ENOENT) {
+
+		fdbg("\ndirectory doesn't exist, we'll create it\n");
 		/* It doesn't exist ... we can create it, but only if we have
 		 * the right permissions and if the parentdirsector is valid. */
 
 		if (parentdirsector == 0xFFFF) {
+			fdbg("\nInvalid parent dir sector\n");
 			/* Invalid entry in the path (non-existant dir segment) */
 
 			goto errout_with_semaphore;
 		}
 
-		/* Check mode */
-
 		/* Create the directory */
 
-		ret = smartfs_createentry(fs, parentdirsector, filename,
-								  SMARTFS_DIRENT_TYPE_DIR, mode, &entry, 0xFFFF, NULL);
+		/* Allocate new data sector for entry */
+		ret = smartfs_alloc_firstsector(&data_sector);
```

Comment:
Is it for first sector ? anyway please think about how we distinguish directory or file

Reply:
Yes, to allocate a first sector for a new entry.
I will have to pass entry type.
Function yet to be written, I'll take care of that.